src/libstd/thread/mod.rs

```diff
@@ -156,6 +194,24 @@ use time::Duration;
 use sys::thread as imp;
 use sys_common::{stack, thread_info};
 
+////////////////////////////////////////////////////////////////////////////////
+// Thread-local storage
+////////////////////////////////////////////////////////////////////////////////
+
+#[macro_use]
+#[doc(hidden)]
+#[unstable(feature = "thread_local_internals")]
+pub mod local;
```

Comment:
Could this be renamed to `__local` to show that you're not supposed to actually access it?

Also, could the unstable tag be placed inside the module itself?